lib/nerves_network/dhcp_manager.ex

```diff
@@ -269,21 +311,18 @@ defmodule Nerves.Network.DHCPManager do
   ## not related to Nerves.Network.Dhclientv4.
   defp consume(:down, :ifadded, state), do: state
 
-  defp consume(:down, :ifremoved, state) do
+  defp consume(:down, reason = :ifremoved, state) do
     state
-    |> stop_dhclient
+    |> stop_dhclient(reason)
     |> goto_context(:removed)
   end
 
   ## Neither of these should be called. Not related to Nerves.Network.Dhclientv4.
   defp consume(:dhcp, {:deconfig, _info}, state), do: state
 
-  defp consume(:dhcp, {:leasefail, _info}, state) do
-    dhcp_retry_timer = Process.send_after(self(), :dhcp_retry, state.dhcp_retry_interval)
-
-    %{state | dhcp_retry_timer: dhcp_retry_timer}
-    |> stop_dhclient
-    |> start_link_local
+  defp consume(:dhcp, {reason = :leasefail, _info}, state) do
```

Comment:
reason here will  be an unused variable, you can prob just remove it if unused.

Reply:
resolved